op-batcher/batcher/channel_manager_test.go

```diff
@@ -726,5 +730,272 @@ func TestChannelManager_TxData_ForcePublish(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        []*types.Block{a, b},
+			batchType:                     derive.SpanBatchType,
+			compressor:                    "",
+			afterAddingToUnsafeBlockQueue: 3813,
+			afterAddingToChannel:          3813,
+			afterSealingChannel:           4590,
+		})
+	})
+
+	t.Run("case8", func(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        []*types.Block{a, b, c},
+			batchType:                     derive.SpanBatchType,
+			compressor:                    "",
+			afterAddingToUnsafeBlockQueue: 5794,
+			afterAddingToChannel:          5794,
+			afterSealingChannel:           6929,
+		})
+	})
+
+	t.Run("case9", func(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        []*types.Block{emptyA},
+			batchType:                     derive.SingularBatchType,
+			compressor:                    "shadow",
+			afterAddingToUnsafeBlockQueue: 70,
+			afterAddingToChannel:          70,
+			afterSealingChannel:           108,
+		})
+	})
+
+	t.Run("case10", func(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        []*types.Block{emptyA, emptyB, emptyC},
+			batchType:                     derive.SingularBatchType,
+			compressor:                    "shadow",
+			afterAddingToUnsafeBlockQueue: 210,
+			afterAddingToChannel:          210,
+			afterSealingChannel:           267,
+		})
+	})
+
+	t.Run("case11", func(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        []*types.Block{emptyA},
+			batchType:                     derive.SpanBatchType,
+			compressor:                    "",
+			afterAddingToUnsafeBlockQueue: 70,
+			afterAddingToChannel:          70,
+			afterSealingChannel:           79,
+		})
+	})
+
+	t.Run("case12", func(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        []*types.Block{emptyA, emptyB, emptyC},
+			batchType:                     derive.SpanBatchType,
+			compressor:                    "",
+			afterAddingToUnsafeBlockQueue: 210,
+			afterAddingToChannel:          210,
+			afterSealingChannel:           81,
+		})
+	})
+
+	t.Run("case13", func(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        twentyBlocks,
+			batchType:                     derive.SingularBatchType,
+			compressor:                    "shadow",
+			afterAddingToUnsafeBlockQueue: 103070,
+			afterAddingToChannel:          103070,
+			afterSealingChannel:           128120,
+		})
+	})
+
+	t.Run("case14", func(t *testing.T) {
+		testChannelManagerUnsafeBytes(t, testCase{
+			blocks:                        tenBlocks,
+			batchType:                     derive.SpanBatchType,
+			compressor:                    "",
+			afterAddingToUnsafeBlockQueue: 50971,
+			afterAddingToChannel:          50971,
+			afterSealingChannel:           61869,
```

Comment:
So the compression in the channel is worse than the DA size estimate? Is that because the tenBlocks have random data and so the size estimator under-estimates?

Looks like this is the case for almost every test case.

Reply:
Yes I think that is likely the reason. In future work we could look into changing the estimate to look at a more worst case estimation. But at least we are starting to track the accuracy in this test.